index.js

```diff
@@ -54,14 +54,48 @@ module.exports = {
 
           const navigate = Page.enable()
             .then(() => Page.navigate({ url }))
+            .then(() => Page.addScriptToEvaluateOnNewDocument({source: `
+              window.addEventListener('application', ({ detail: Application }) => {
+                Application.autoboot = false;
+              });
+            `}))
             .then(() => Page.loadEventFired());
 
           return navigate
             .then(() => Runtime.evaluate({ awaitPromise: true, expression: `
-              ${this._appGlobal()}.visit('${visitPath}')
-                .then((application) => {
-                  return document.body.querySelector('.ember-view').outerHTML;
-                });
+              // Wrap inside a native promise since appGlobal.visit returns
+              // RSVP.Promise which doesn't play well with awaitPromise.
+              new Promise((resolve, reject) => {
+                const getOuterHTML = () => {
+                  try {
+                    window['${this._appGlobal()}'].visit('${visitPath}')
+                      .then((application) => {
+                        return resolve(document.body.querySelector('.ember-view').outerHTML);
+                      });
+                  } catch(e) {
+                    // Log errors for future debugging.
+                    return reject(e);
+                  }
+                };
+                if (window['${this._appGlobal()}']) {
+                  getOuterHTML();
+                } else {
```

Comment:
I'm not exactly following the logic here. It looks like you have edge cases with the global is not defined and you are creating it yourself?

Reply:
Exactly. The appGlobal will then be stored inside `this._appGlobal`, but since the getter is defined `window[appGloblal]` will return it.

For example given:
```
window['Foo']
undefined
const doSomethingWithFoo = () => {
	console.log('foo is', window['Foo']);
};
doSomethingWithFoo();
foo is undefined
Object.defineProperty(window, 'Foo', {
  configurable: true,
  enumerable: true,
  writeable: true,
  get: function() {
    return this._internalFoo;
  },
  set: function(val) {
    this._internalFoo = val;
    doSomethingWithFoo();
  }
});
window.Foo = 'bar';
// after setting the global the setter function will be invoked & doSomethingWithFoo called
foo is bar
```

Reply:
Clever :D